qcengine/programs/orca.py

```diff
@@ -0,0 +1,297 @@
+
+        # Setup the job
+        job_inputs = self.build_input(input_data, config)
+
+        # Run Orca
+        exe_success, proc = self.execute(job_inputs)
+
+        # Determine whether the calculation succeeded
+        if exe_success:
+            # If execution succeeded, collect results
+            result = self.parse_output(proc, input_data)
+            return result
+        else:
+            # Return UnknownError for error propagation
+            return UnknownError(proc["stderr"])
+
+    def execute(
+        self,
+        inputs: Dict[str, Any],
+        extra_infiles: Optional[Dict[str, str]] = None,
+        extra_outfiles: Optional[List[str]] = None,
+        as_binary: Optional[List[str]] = None,
+        extra_commands: Optional[List[str]] = None,
+        scratch_name: Optional[str] = None,
+        scratch_messy: bool = False,
+        timeout: Optional[int] = None,
+    ) -> Tuple[bool, Dict[str, Any]]:
+        """
+        For option documentation go look at qcengine/util.execute
+        """
+        infiles = inputs["infiles"]
+        if extra_infiles is not None:
+            infiles.update(extra_infiles)
+
+        # Collect all output files and update with extra_outfiles
+        outfiles = ["dispatch.engrad"]
+
+        if extra_outfiles is not None:
+            outfiles.extend(extra_outfiles)
+
+        # Replace commands with extra_commands if present
+        commands = inputs["commands"]
+        if extra_commands is not None:
+            commands = extra_commands
+
+        # Run the Orca program
+        exe_success, proc = execute(
+            commands,
+            infiles=infiles,
+            outfiles=outfiles,
+            as_binary=as_binary,
+            scratch_name=scratch_name,
+            scratch_directory=inputs["scratch_directory"],
+            scratch_messy=scratch_messy,
+            timeout=timeout,
+        )
+        return exe_success, proc
+
+    def build_input(
+        self, input_model: "AtomicInput", config: "JobConfig", template: Optional[str] = None
+    ) -> Dict[str, Any]:
+
+        if template is None:
+            input_file = []
+
+            # Resolving keywords
+            # caseless_keywords = {k.lower(): v for k, v in input_model.keywords.items()}
+
+            # Write the geom
+            xyz_block = input_model.molecule.to_string(dtype="orca", units="Angstrom")
+
+            # Determine what SCF type (restricted vs. unrestricted)
+            hf_type = "RHF"
+            # dft_type = "RKS"
+            # if unrestricted:
+            #     hf_type = "UHF"
+            #     dft_type = "UKS"
+
+            # Write energy call
+            energy_call = []
+
+            if input_model.model.method.upper() in self._post_hf_methods:  # post SCF case
+                energy_call.append(f"{{{hf_type}}}")
+                energy_call.append("")
+                energy_call.append(f"{{{input_model.model.method}}}")
```

Comment:
would a post-scf ever get the coordinates or a basis set?

Reply:
I will create some inputs for post HF and update accordingly. I was just playing with HF and DFT.